compiler/src/main/java/com/squareup/anvil/compiler/codegen/CodeGenerationExtension.kt

```diff
@@ -47,7 +49,7 @@ internal class CodeGenerationExtension(
   ): AnalysisResult? {
     // Tell the compiler that we have something to do in the analysisCompleted() method if
     // necessary.
-    return if (!didRecompile && codeGenerators.isNotEmpty()) AnalysisResult.EMPTY else null
```

Comment:
We know that `codeGenerators` is never empty. That check was unnecessary.

Reply:
> We know that codeGenerators is never empty. That check was unnecessary.

Is this actually guaranteed? From looking at the constructor and field initialization, it doesn't seem like this class knows that (even if we might know it outside of this class for its usage).

Reply:
We know that, because within Anvil itself are plenty of code generators. It's okay for this class to have some assumptions about the internals of the compiler plugin.